agot-bg-game-server/src/common/ingame-game-state/game-data-structure/house-card/StannisBaratheonDwDHouseCardAbility.ts

```diff
@@ -0,0 +1,38 @@
+import HouseCardAbility from "./HouseCardAbility";
+import House from "../House";
+import HouseCard from "./HouseCard";
+import CancelHouseCardAbilitiesGameState
+    from "../../action-game-state/resolve-march-order-game-state/combat-game-state/cancel-house-card-abilities-game-state/CancelHouseCardAbilitiesGameState";
+
+export default class StannisBaratheonDwDHouseCardAbility extends HouseCardAbility {
+
+    cancel(cancelResolutionState: CancelHouseCardAbilitiesGameState, house: House, _houseCard: HouseCard): void {
```

Comment:
Yes, I agree with you. I was not thinking about this scenario when I was implementing this card. This card is very similar to Walder Frey and it should execute after 'cancel' and 'immediately' phases.

Also, now when I'm thinking about this card I maybe see another problem.
Should we remove support from lands when the players are fighting on the sea? Because, a land never can support a fight on sea, the support can't be _provided_

Now, the implementation is removing every supports, without checking if the battle is on land or sea. What do you think? Should it be fixed?

Reply:
I guess it’s implemented correctly. The card text says „remove all support orders from adjacent regions“. Though a land may never support a sea battle it is adjacent to that area.

Reply:
On the other hand Stannis is also no immediately card as the term immediately is not used. Instead it says cancel support, so we can consider it as cancel card i think and keep it as it is...

Reply:
After a discussion on Discord it’s clear that we have to adapt Stannis. Faq says cancel abilities start with the wording cancel and so only tyrion and qot dwd are real cancel cards.